src/include/sof/audio/audio_stream.h

```diff
@@ -1041,9 +1041,15 @@ static inline void audio_stream_fmt_conversion(enum ipc4_bit_depth depth,
 	*valid_fmt = (valid >> 3) - 2;
 
 	/* really 24_3LE */
-	if (valid == 24 && depth == 24) {
-		*frame_fmt = SOF_IPC_FRAME_S24_3LE;
-		*valid_fmt = SOF_IPC_FRAME_S24_3LE;
+	if (valid == 24) {
+		if (depth == 24) {
+			*frame_fmt = SOF_IPC_FRAME_S24_3LE;
+			*valid_fmt = SOF_IPC_FRAME_S24_3LE;
+		} else if (type == IPC4_TYPE_MSB_INTEGER) {
+			*valid_fmt = SOF_IPC_FRAME_S24_4LE_MSB;
+		}
```

Comment:
Should we have a case for depth==24,valid==24,type=LSB_INTEGER (-> SOF_IPC_FRAME_S24_LE)

Reply:
depth==24,valid==24 it means packed 24bits format. This is required by windows and not supported by Linux now

Reply:
Sorry, my mistake, I meant "depth==32,valid==24,type=LSB_INTEGER -> SOF_IPC_FRAME_S24_LE".

Reply:
@kv2019i  it is done before this setting
````c
	/* IPC4_DEPTH_16BIT (16) <---> SOF_IPC_FRAME_S16_LE (0)
	 * IPC4_DEPTH_24BIT (24) <---> SOF_IPC_FRAME_S24_4LE (1)
	 * IPC4_DEPTH_32BIT (32) <---> SOF_IPC_FRAME_S32_LE (2)
	 */
	*frame_fmt = (depth >> 3) - 2;
	*valid_fmt = (valid >> 3) - 2; // convert to SOF_IPC_FRAME_S24_LE

	/* really 24_3LE */
	if (valid == 24 && depth == 24) {
		*frame_fmt = SOF_IPC_FRAME_S24_3LE;
		*valid_fmt = SOF_IPC_FRAME_S24_3LE;
	}
````